Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -364,30 +321,43 @@ WarpX::computePhi (const amrex::Vector<std::unique_ptr<amrex::MultiFab> >& rho,
             info.setMaxSemicoarseningLevel(max_semicoarsening_level);
             info.setSemicoarseningDirection(semicoarsening_direction);
         }
-        MLNodeTensorLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info );
-        linop.setBeta( beta_solver );
+
+#if defined(AMREX_USE_EB) || defined(WARPX_DIM_RZ)
+        // In the presence of EB or RZ: the solver assumes that the beam is
+        // propagating along  one of the axes of the grid, i.e. that only *one*
+        // of the components of `beta` is non-negligible.
+        MLEBNodeFDLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info
+#if defined(AMREX_USE_EB)
+            , {&WarpX::fieldEBFactory(lev)}
 #endif
-#else
-        // With embedded boundary: extract EB info
-        MLEBNodeFDLaplacian linop( {Geom(lev)}, {boxArray(lev)}, {DistributionMap(lev)}, info, {&WarpX::fieldEBFactory(lev)});
+        );
```

Comment:
Got the backtrace:
```
0::Assertion `i < (this->std::vector<T, Allocator>::size())' failed, file "/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_Vector.H", line 42 !!!
SIGABRT
```
```
=== If no file names and line numbers are shown below, one can run
            addr2line -Cpfie my_exefile my_line_address
    to convert `my_line_address` (e.g., 0x4a6b) into file name and line number.
    Or one can use amrex/Tools/Backtrace/parse_bt.py.

=== Please note that the line number reported by addr2line may not be accurate.
    One can use
            readelf -wl my_exefile | grep my_line_address'
    to find out the offset for that line.

 0: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x864886) [0x557f0a312886]
    amrex::BLBackTrace::print_backtrace_info(_IO_FILE*)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_BLBackTrace.cpp:175:29

 1: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x8642a1) [0x557f0a3122a1]
    amrex::BLBackTrace::handler(int)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_BLBackTrace.cpp:85:15

 2: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x73a9f4) [0x557f0a1e89f4]
    amrex::ParallelDescriptor::Abort(int, bool)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_ParallelDescriptor.cpp:206:29

 3: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x6f2319) [0x557f0a1a0319]
    amrex::Assert_host(char const*, char const*, int, char const*)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX.cpp:270:9

 4: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x3859fa) [0x557f09e339fa]
    amrex::Assert(char const*, char const*, int, char const*) inlined at /home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_Vector.H:42:9 in amrex::Vector<amrex::EB2::GShopLevel<amrex::EB2::GeometryShop<(anonymous namespace)::ParserIF, amrex::Parser> >, std::allocator<amrex::EB2::GShopLevel<amrex::EB2::GeometryShop<(anonymous namespace)::ParserIF, amrex::Parser> > > >::operator[](unsigned long) const
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX.H:185:5
operator[]
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_Vector.H:42:9

 5: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x38589d) [0x557f09e3389d]
    getLevel
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/EB/AMReX_EB2_IndexSpaceI.H:69:1

 6: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x8f8f1e) [0x557f0a3a6f1e]
    amrex::makeEBFabFactory(amrex::Geometry const&, amrex::BoxArray const&, amrex::DistributionMapping const&, amrex::Vector<int, std::allocator<int> > const&, amrex::EBSupport)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/EB/AMReX_EBFabFactory.cpp:119:61

 7: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xa53ae7) [0x557f0a501ae7]
    amrex::MLEBNodeFDLaplacian::makeFactory(int, int) const
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLEBNodeFDLaplacian.cpp:133:53

 8: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xa42088) [0x557f0a4f0088]
    amrex::MLLinOp::defineGrids(amrex::Vector<amrex::Geometry, std::allocator<amrex::Geometry> > const&, amrex::Vector<amrex::BoxArray, std::allocator<amrex::BoxArray> > const&, amrex::Vector<amrex::DistributionMapping, std::allocator<amrex::DistributionMapping> > const&, amrex::Vector<amrex::FabFactory<amrex::FArrayBox> const*, std::allocator<amrex::FabFactory<amrex::FArrayBox> const*> > const&)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLLinOp.cpp:521:69

 9: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xa38cd3) [0x557f0a4e6cd3]
    amrex::MLLinOp::define(amrex::Vector<amrex::Geometry, std::allocator<amrex::Geometry> > const&, amrex::Vector<amrex::BoxArray, std::allocator<amrex::BoxArray> > const&, amrex::Vector<amrex::DistributionMapping, std::allocator<amrex::DistributionMapping> > const&, amrex::LPInfo const&, amrex::Vector<amrex::FabFactory<amrex::FArrayBox> const*, std::allocator<amrex::FabFactory<amrex::FArrayBox> const*> > const&, bool)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLLinOp.cpp:169:18

10: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xac9f93) [0x557f0a577f93]
    amrex::MLNodeLinOp::define(amrex::Vector<amrex::Geometry, std::allocator<amrex::Geometry> > const&, amrex::Vector<amrex::BoxArray, std::allocator<amrex::BoxArray> > const&, amrex::Vector<amrex::DistributionMapping, std::allocator<amrex::DistributionMapping> > const&, amrex::LPInfo const&, amrex::Vector<amrex::FabFactory<amrex::FArrayBox> const*, std::allocator<amrex::FabFactory<amrex::FArrayBox> const*> > const&, int)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLNodeLinOp.cpp:38:20

11: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xa536ed) [0x557f0a5016ed]
    amrex::MLEBNodeFDLaplacian::define(amrex::Vector<amrex::Geometry, std::allocator<amrex::Geometry> > const&, amrex::Vector<amrex::BoxArray, std::allocator<amrex::BoxArray> > const&, amrex::Vector<amrex::DistributionMapping, std::allocator<amrex::DistributionMapping> > const&, amrex::LPInfo const&, amrex::Vector<amrex::EBFArrayBoxFactory const*, std::allocator<amrex::EBFArrayBoxFactory const*> > const&)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLEBNodeFDLaplacian.cpp:97:24

12: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0xa5320d) [0x557f0a50120d]
    amrex::MLEBNodeFDLaplacian::MLEBNodeFDLaplacian(amrex::Vector<amrex::Geometry, std::allocator<amrex::Geometry> > const&, amrex::Vector<amrex::BoxArray, std::allocator<amrex::BoxArray> > const&, amrex::Vector<amrex::DistributionMapping, std::allocator<amrex::DistributionMapping> > const&, amrex::LPInfo const&, amrex::Vector<amrex::EBFArrayBoxFactory const*, std::allocator<amrex::EBFArrayBoxFactory const*> > const&)
/home/axel/src/warpx/build/_deps/fetchedamrex-src/Src/LinearSolvers/MLMG/AMReX_MLEBNodeFDLaplacian.cpp:18:1

13: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x3a7407) [0x557f09e55407]
    WarpX::computePhi(amrex::Vector<std::unique_ptr<amrex::MultiFab, std::default_delete<amrex::MultiFab> >, std::allocator<std::unique_ptr<amrex::MultiFab, std::default_delete<amrex::MultiFab> > > > const&, amrex::Vector<std::unique_ptr<amrex::MultiFab, std::default_delete<amrex::MultiFab> >, std::allocator<std::unique_ptr<amrex::MultiFab, std::default_delete<amrex::MultiFab> > > >&, std::array<double, 3ul>, double, double, int, int) const
/home/axel/src/warpx/Source/FieldSolver/ElectrostaticSolver.cpp:333:9

14: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x3a4a72) [0x557f09e52a72]
    WarpX::AddSpaceChargeFieldLabFrame()
/home/axel/src/warpx/Source/FieldSolver/ElectrostaticSolver.cpp:234:20

15: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x3a37b2) [0x557f09e517b2]
    WarpX::ComputeSpaceChargeField(bool)
/home/axel/src/warpx/Source/FieldSolver/ElectrostaticSolver.cpp:67:36

16: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x41decc) [0x557f09ecbecc]
    WarpX::InitData()
/home/axel/src/warpx/Source/Initialization/WarpXInitData.cpp:397:44

17: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x4c9e8) [0x557f09afa9e8]
    main
/home/axel/src/warpx/Source/main.cpp:65:21

18: /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fd0399560b3]
    __libc_start_main
../csu/libc-start.c:342:3

19: ./warpx.RZ.MPI.OMP.DP.OPMD.EB.QED.DEBUG(+0x4c82e) [0x557f09afa82e]
    ?? ??:0


===== TinyProfilers ======
main()
WarpX::InitData()
WarpX::ComputeSpaceChargeField
WarpX::AddSpaceChargeFieldLabFrame
MLEBNodeFDLaplacian::define()
MLLinOp::define()
MLLinOp::defineGrids()
```

Reply:
https://github.com/AMReX-Codes/amrex/blob/3ca867b8cce86b9892b45ff78df8630500dc3bb4/Src/EB/AMReX_EBFabFactory.cpp#L119

Reply:
Is it possible that we cannot construct the `MLEBNodeFDLaplacian` class with a single level and need to pass all levels at once, if we also pass the `a_factory` argument? @WeiqunZhang ?

Reply:
No, looks like it's something else.  How to reproduce the error?

Reply:
I don't know if this has been answered elsewhere (on Slack, offline, etc.), but I would reproduce it in the following way:
- Add `-DCMAKE_BUILD_TYPE=Debug` to the `cmakeSetupOpts` of the test `ElectrostaticSphereEB_RZ` in Regression/WarpX-tests.ini.
- Run `./run_test.sh ElectrostaticSphereEB_RZ` locally.
- Inspect standard output and various log files saved in `test_dir/rt-WarpX/WarpX-tests/2022-05-13/ElectrostaticSphereEB_RZ/` (note that the date in this path could need to be updated):
```
Backtrace.0.0  ElectrostaticSphereEB_RZ.err.out  ElectrostaticSphereEB_RZ.make.out  inputs_rz
Backtrace.1.0  ElectrostaticSphereEB_RZ.ex       ElectrostaticSphereEB_RZ.run.out
```
- Then you can also take the input file and run it on its own, without `run_test.sh`.